### PR TITLE
feat(ext): expand scopes and expand when parsing

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: rs-lib/src/lib.rs [hunk 1]
```diff
@@ -565,40 +578,6 @@ impl ImportMap {
...
-        .insert(key.to_string(), Some(value.as_str().unwrap().to_string()));
-    }
-    let mut diagnostics = vec![];
-    let imports = parse_specifier_map(
-      expanded_imports_im,
-      &self.base_url,
-      &mut diagnostics,
-    );
-    self.imports = imports;
-  }
```
Comment (dsherret) [inline rs-lib/src/lib.rs:601]: It was starting to get complicated making this work. I realized we can just do this when parsing, which is more efficient and less code to make this work.

Comment (bartlomieju) [reply rs-lib/src/lib.rs:601]: I don't we should do it during parsing - we only want that for `deno.json` and `deno.jsonc` files - not the import map from `--import-map` flag.

Comment (bartlomieju) [reply rs-lib/src/lib.rs:601]: Nvm, I see you added an option.

Comment (dsherret) [reply rs-lib/src/lib.rs:601]: Yeah, deno_config will be able to handle it easily https://github.com/denoland/deno_config/pull/41

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**